icechunk/src/config.rs

```diff
@@ -22,17 +22,19 @@ pub struct S3Options {
     pub endpoint_url: Option<String>,
     pub anonymous: bool,
     pub allow_http: bool,
+    pub force_path_style: Option<bool>,
```

Comment:
Can we keep this a `bool` but instruct serde to use "false" if not found? I think we can do that. we need to be very careful with backwards compatibility

Reply:
The current behaviour is to force path style when `allow_http` is True, so i think we nee d an Option?

Reply:
I think we can break that behavior in favor of the simplicity of a bool? What do you think?

In any case, we'll need a test to check that we can still read a persisted old configuration, because I think this class goes into the virtual chunks configuration. I think if you do nothing it won't be backwards compatible, you probably need to flag it as "use default if not present".

I'd also add a comment explaining on what version the field was added

Reply:
I used `test_can_read_old`. Is that enough?

Reply:
i think test can read is enough, because it has a config file with S3Options.